gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -267,6 +270,7 @@ private Completable handlePhotoCaptureResult(String fieldId) {
             })
         .doOnNext(
             url -> {
+              isPhotoFieldUpdated = true;
```

Comment:
May be out of scope of this PR, but it would seem that saving the bitmap should be an asynchronous operation, whereas here it's modeled as a synchronous one. 

Also, the Rx chain seems to be relying on side effects to do critical work, which is general not harder to debug and maintain. It might be clearer if this method would accept `Bitmap` instead of an `Observable<Bitmap>`, save and upload asynchronously, and return a `Completable` or `Single<String>` when done. Wdyt?

Reply:
We're handling the result of the activity before the state of the response is reset. To fix this, we should call the external intent from the activity and handle for results in `onObservationLoaded` after `refreshResponseMap` call.

Reply:
See comment below re `isPhotoFieldUpdated`.

Could you kindly collapse the chained `map` and `doOnNext` into a single function and use a more appropriate Rx method for chaining? Might look something like this:

```java
  private Completable handlePhotoCaptureResult(String fieldId) {
    return cameraManager
        .capturePhotoResult()
        .flatMapCompletable(bitmap -> saveBitmapAndUpdateResponse(bitmap, fieldId));
  }

  private Completable saveBitmapAndUpdateResponse(
      Bitmap bitmap, String fieldId) {
    String file = fileUtil.saveBitmap(bitmap, fieldId + ".jpg");
    String destinationPath = getRemoteImagePath(file.getName());
    String url = firestoreStorageManager.uploadMediaFromFile(file, destinationPath);
    isPhotoFieldUpdated = true;
    onTextChanged(form.getValue().getField(fieldId).get(), url);
    return Complete.complete()
  }
```

I assume we'd want `capturePhotoResult()` to run on the UI thread, and `saveBitmapAndUpdateResponse()` to run in the IO thread. @dturner, what's the right way to do this? I vaguely remember downstream `flatMap` logic runs on the same thread as the upstream subscribers (defined with `subscribeOn`), while `switchMap` does not, but I can't find a reference for it.